packages/lit-auth-client/src/lib/lit-auth-client.ts

```diff
@@ -82,385 +65,102 @@ export class LitAuthClient {
-      window.document.location.search
-    );
-    // Check if current url is redirect uri and has redirect params
-    if (
-      isRedirectUri &&
-      (provider || accessToken || idToken || state || error)
-    ) {
-      return true;
-    }
-    return false;
+  initGoogleProvider(params: InitOAuthProviderOptions): GoogleProvider {
+    const provider = new GoogleProvider({
+      ...params,
+      rpcUrl: this.rpcUrl,
+      relay: this.relay,
+      litNodeClient: this.litNodeClient,
+    });
+    this.providers.set('google', provider);
+    return provider;
   }
 
   /**
-   * Generate a wallet signature to use as an auth method
+   * Initialize a Discord OAuth provider
    *
-   * @param {SignInWithEthWalletParams} params
-   * @param {string} params.address - Ethereum wallet address
-   * @param {Function} params.signMessage - Function to sign message
-   * @param {string} [params.origin] - Origin of signing request
-   * @param {string} [params.chain] - Name of chain to use for signature
-   * @param {number} [params.expiration] - When the auth signature expires
+   * @param {InitOAuthProviderOptions} params
+   * @param {string} params.redirectUri - Redirect URI that Lit's login server should send the user back to
    *
-   * @returns {AuthMethod} - Auth method object containing the auth signature
+   * @returns {DiscordProvider} - Discord OAuth provider
    */
-  public async signInWithEthWallet(
-    params: SignInWithEthWalletParams
-  ): Promise<AuthMethod> {
-    // Get chain ID or default to Ethereum mainnet
-    const chain = params.chain || 'ethereum';
-    const selectedChain = LIT_CHAINS[chain];
-    const chainId = selectedChain?.chainId ? selectedChain.chainId : 1;
-
-    // Get expiration or default to 24 hours
-    const expiration =
-      params.expiration ||
-      new Date(Date.now() + 1000 * 60 * 60 * 24).toISOString();
-
-    // Prepare Sign in with Ethereum message
-    const preparedMessage: Partial<SiweMessage> = {
-      domain: this.domain,
-      address: ethers.utils.getAddress(params.address), // convert to EIP-55 format or else SIWE complains
-      version: '1',
-      chainId,
-      expirationTime: expiration,
-    };
-
-    if (origin) {
-      preparedMessage.uri = origin;
-    } else {
-      preparedMessage.uri = globalThis.location.href;
-    }
-
-    const message: SiweMessage = new SiweMessage(preparedMessage);
-    const toSign: string = message.prepareMessage();
-
-    // Use provided function to sign message
-    const signature = await params.signMessage(toSign);
-
-    const authSig: AuthSig = {
-      sig: signature,
-      derivedVia: 'web3.eth.personal.sign',
-      signedMessage: toSign,
-      address: params.address,
-    };
-
-    const authMethod = {
-      authMethodType: AuthMethodType.EthWallet,
-      accessToken: JSON.stringify(authSig),
-    };
-    return authMethod;
+  initDiscordProvider(params: InitOAuthProviderOptions): DiscordProvider {
```

Comment:
Perhaps this shouldn't be explicitly defined for each provider type.
maybe instead there is a single `initProvider<T extends ProviderOptions>(params: ProviderOptions<T>)`
T in the `ProviderOptions` will be a concrete interface type for the Provider Options. A bit cleaner than having an explicit init.

Reply:
@joshLong145 - updated, thanks!